File: src/leetcode/java/medium/GrayCode_89.java
package leetcode.java.medium;

import java.util.ArrayList;
import java.util.List;

public class GrayCode_89 {
  public List<Integer> grayCode(int n) {
    List<Integer> res = new ArrayList<>(1 << n);
    for (int i = 0; i < 1 << n; i++) res.add(i ^ (i >> 1));
    return res;
  }
}

File: src/leetcode/python/GrayCode_89.py
class GrayCode_89(object):
    def grayCode(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        res = [0] * (1 << n)
        for i in range(len(res)):
            res[i] = i ^ (i >> 1)
        return res
